src/c2goto/headers/ubuntu20.04/kernel_5.15.0-76/include/linux/spinlock.h

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#include <stdatomic.h>
+/* The declarations of mock spin lock behaviour*/
+/** mock spinlock struct*/
+typedef struct {
+    int locked;
```

Comment:
Replace `int` with `bool` or maybe:
```c
typedef enum {
  UNLOCKED,
  LOCKED
} State;

typedef struct {
  State state;
} spinlock_t;
```

Reply:
Well spotted!

Remember that we convert the underlying logical formula to bits!

So, if we can reduce the number of bits, we will help the SAT solver!